src/base_alloc/base_alloc.c

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+*/
+
+#include <assert.h>
+
+#include "base_alloc.h"
+#include "base_alloc_internal.h"
+#include "utils_common.h"
+#include "utils_concurrency.h"
+
+// minimum size of a single pool of the base allocator,
+#define MINIMUM_POOL_SIZE (ba_os_get_page_size())
+
+// minimum number of chunks used to calculate the size of pools
+#define MINIMUM_CHUNK_COUNT (128)
+
+// alignment of the base allocator
+#define MEMORY_ALIGNMENT (8)
+
+typedef struct ba_mem_chunk_t ba_mem_chunk_t;
+
+// memory chunk of size 'chunk_size'
+struct ba_mem_chunk_t {
+    ba_mem_chunk_t *next;
+};
+
+struct umf_ba_pool_meta_t {
+    // address of the beginning of this pool (allocated region)
+    umf_ba_pool_t *pool_addr;
```

Comment:
why this is umf_ba_pool_t and not umf_ba_pool_meta_t? You got a circular dependency

Reply:
It is not a circular dependency, It is only a pointer stored, but actually it is not necessary, so I removed it.
Done.